optuna/cli.py

```diff
@@ -855,16 +862,26 @@ def take_action(self, parsed_args: Namespace) -> int:
         )
 
         state = TrialState[parsed_args.state.upper()]
-        study.tell(
-            trial=parsed_args.trial_number,
-            values=parsed_args.values,
-            state=state,
-        )
+        trial_number = parsed_args.trial_number
+        values = parsed_args.values
+
+        # TODO(hvy): Consider introducing `skip_if_finished` to `Study.tell` and let the method
+        # do all error handling, such as invalid trial numbers.
+        if parsed_args.skip_if_finished:
+            trials = study.get_trials(deepcopy=False)
+            trial = trials[trial_number]
+            assert trial.number == trial_number
+
+            if trial.state.is_finished():
+                self.logger.info(
+                    f"Skipped telling trial {trial_number} with values "
+                    f"{values} and state {state} since trial was already finished."
+                )
```

Comment:
How about logging `trial.state` and `trial.values` as well as `state` and `values`?

Reply:
Sounds good. I'll fix that.

Reply:
Just fixed, PTAL if you have time over 🙇

Reply:
Thank you for the swift fix!